tests/test_updater_ng.py

```diff
@@ -336,6 +336,16 @@ def test_persist_metadata_fails(
         for filename in os.listdir(self.updater._dir):
             self.assertFalse(filename.startswith("tmp"))
 
+    def test_invalid_target_base_url(self) -> None:
+        info = TargetFile(1, {"sha256": ""}, "targetpath")
+        with self.assertRaises(exceptions.URLParsingError):
+            self.updater.download_target(info, target_base_url="invalid_url")
+
+    def test_non_existing_target_file(self) -> None:
+        info = TargetFile(1, {"sha256": ""}, "")
+        with self.assertRaises(exceptions.DownloadLengthMismatchError):
+            self.updater.download_target(info)
```

Comment:
the error changed here without explanation...

Reply:
@jku I separated to two different tests to cover both cases with a description of the `DownloadLengthMismatchError`.  The difference comes after I moved to `test_updater_ng` with using a webserver rather than a simulator. In this case dir listing is returned for empty path, so it results to the mismatch error, while with the simulator such listing cannot be returned, as a real fetch doesn't happen actually, so it was failing with `no target` exception. I added one more test for non-existing target file to cover the previous use-case.

Reply:
This is now testing a peculiarity of the webserver we use: it returns a directory listing if you ask for a directory. So the reason for the error is not really the empty filepath but how the webserver reacts in this specific case... I think this test is just reproducing test_length_hash_mismatch() so is not useful.